libtextsecure/message_receiver.js

```diff
@@ -1065,6 +1065,14 @@ MessageReceiver.prototype.extend({
       promises.push(this.handleAttachment(attachment));
     }
 
+    if (
+      decrypted.contact &&
+      decrypted.contact.avatar &&
+      decrypted.contact.avatar.avatar
```

Comment:
Oh man, now I see it! `contact` is an array 😢 Can’t wait to build our own parsing layer that avoids problems like this due to naming. I bet the other platforms don‘t run into that because they get compiler support.